containsExactly creates ArrayIndexOutOfBoundsException

This test case reproduces the error:
```
    @Test
    fun foo() {
        assertThat(listOf("1", "2", "3")).containsExactly("2", "3", "1")
    }
```
Both lists are of same size, contain the same elements in the same order except on the first element of the `actual` list which is expected to be at the end of the list:

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: -1
	at assertk.assertions.support.ListDiffer.backtrack(ListDiffer.kt:87)
	at assertk.assertions.support.ListDiffer.diff(ListDiffer.kt:10)
	at assertk.assertions.ListKt.listDifferExpected(list.kt:48)
	at assertk.assertions.ListKt.containsExactly(list.kt:44)
	... 24 more
```

tests with assertk 0.13

--- a/assertk-common/src/main/kotlin/assertk/assertions/support/ListDiffer.kt
+++ b/assertk-common/src/main/kotlin/assertk/assertions/support/ListDiffer.kt
@@ -84,7 +84,7 @@ internal object ListDiffer {
                 k - 1
             }
 
-            val prevX = v[prevK]
+            val prevX = v.ringIndex(prevK)
             val prevY = prevX - prevK
 
             while (x > prevX && y > prevY) {

--- a/assertk-common/src/test/kotlin/test/assertk/assertions/ListTest.kt
+++ b/assertk-common/src/test/kotlin/test/assertk/assertions/ListTest.kt
@@ -26,6 +26,19 @@ class ListTest {
         )
     }
 
+    // https://github.com/willowtreeapps/assertk/issues/185
+    @Test fun containsExactly_elements_in_different_order_fails2() {
+        val error = assertFails {
+            assertThat(listOf("1", "2", "3")).containsExactly("2", "3", "1")
+        }
+        assertEquals(
+            """expected to contain exactly:
+                | at index:0 unexpected:<"1">
+                | at index:2 expected:<"1">
+            """.trimMargin(), error.message
+        )
+    }
+
     @Test fun containsExactly_same_indexes_are_together() {
         val error = assertFails {
             assertThat(listOf(1, 1)).containsExactly(2, 2)

